shell/platform/linux/fl_method_response.cc

```diff
@@ -0,0 +1,172 @@
+  FlMethodResponse parent_instance;
+
+  FlValue* result;
+};
+
+struct _FlMethodErrorResponse {
+  FlMethodResponse parent_instance;
+
+  gchar* code;
+  gchar* message;
+  FlValue* details;
+};
+
+struct _FlMethodNotImplementedResponse {
+  FlMethodResponse parent_instance;
+};
+
+G_DEFINE_TYPE(FlMethodResponse, fl_method_response, G_TYPE_OBJECT)
+G_DEFINE_TYPE(FlMethodSuccessResponse,
+              fl_method_success_response,
+              fl_method_response_get_type())
+G_DEFINE_TYPE(FlMethodErrorResponse,
+              fl_method_error_response,
+              fl_method_response_get_type())
+G_DEFINE_TYPE(FlMethodNotImplementedResponse,
+              fl_method_not_implemented_response,
+              fl_method_response_get_type())
+
+static void fl_method_response_class_init(FlMethodResponseClass* klass) {}
+
+static void fl_method_response_init(FlMethodResponse* self) {}
+
+static void fl_method_success_response_dispose(GObject* object) {
+  FlMethodSuccessResponse* self = FL_METHOD_SUCCESS_RESPONSE(object);
+
+  g_clear_pointer(&self->result, fl_value_unref);
+
+  G_OBJECT_CLASS(fl_method_success_response_parent_class)->dispose(object);
+}
+
+static void fl_method_success_response_class_init(
+    FlMethodSuccessResponseClass* klass) {
+  G_OBJECT_CLASS(klass)->dispose = fl_method_success_response_dispose;
+}
+
+static void fl_method_success_response_init(FlMethodSuccessResponse* self) {}
+
+static void fl_method_error_response_dispose(GObject* object) {
+  FlMethodErrorResponse* self = FL_METHOD_ERROR_RESPONSE(object);
+
+  g_clear_pointer(&self->code, g_free);
+  g_clear_pointer(&self->message, g_free);
+  g_clear_pointer(&self->details, fl_value_unref);
+
+  G_OBJECT_CLASS(fl_method_error_response_parent_class)->dispose(object);
+}
+
+static void fl_method_error_response_class_init(
+    FlMethodErrorResponseClass* klass) {
+  G_OBJECT_CLASS(klass)->dispose = fl_method_error_response_dispose;
+}
+
+static void fl_method_error_response_init(FlMethodErrorResponse* self) {}
+
+static void fl_method_not_implemented_response_class_init(
+    FlMethodNotImplementedResponseClass* klass) {}
+
+static void fl_method_not_implemented_response_init(
+    FlMethodNotImplementedResponse* self) {}
+
+G_MODULE_EXPORT FlValue* fl_method_response_get_result(FlMethodResponse* self,
+                                                       GError** error) {
+  if (FL_IS_METHOD_SUCCESS_RESPONSE(self))
+    return fl_method_success_response_get_result(
+        FL_METHOD_SUCCESS_RESPONSE(self));
+
+  if (FL_IS_METHOD_ERROR_RESPONSE(self)) {
+    const gchar* code =
+        fl_method_error_response_get_code(FL_METHOD_ERROR_RESPONSE(self));
+    const gchar* message =
+        fl_method_error_response_get_message(FL_METHOD_ERROR_RESPONSE(self));
+    // FIXME(robert-ancell) Encode the error_details with JSON and show in the
+    // message FlValue *details =
+    // fl_method_error_response_get_details(FL_METHOD_ERROR_RESPONSE(self));
```

Comment:
We shouldn't land commented-out code. Why can't this be implemented as part of the PR?

Reply:
At the time the JSON codec wasn't landed, but I can implement this now.